### PR TITLE
Hide mtDNA and coverage reports for WTS cases

### LINKED ISSUE #4639: Hide not-applicable reports for e.g. tomte cases
If no coverage or mito data is available from DNA pipes, hiding the report links would be desirable.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4639

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested on stage: 🆗 

image

File: tests/server/test_server_utils.py [hunk 1]
```diff
@@ -62,6 +64,34 @@ def test_case_has_mt_alignments(case_obj):
     assert case_obj["mt_bams"] is True
 
 
+def test_case_has_case_has_mtdna_report_show(case_obj):
```
Comment (dnil) [inline tests/server/test_server_utils.py:67]: 👍

**Review by dnil — APPROVED** [review]
Nice, thank you!

File: scout/server/utils.py [hunk 2]
```diff
@@ -206,12 +206,24 @@ def user_institutes(store, login_user):
     return institutes
 
 
+def case_has_mtdna_report(case_obj: dict):
+    """Display mtDNA report on the case sidebar only for some specific cases."""
+    if case_obj.get("track", "rare") != "cancer":
+        for ind in case_obj.get("individuals", []):
+            if ind.get("analysis_type") != "wts":
+                case_obj["mtdna_report"] = True
+                return
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply scout/server/utils.py:215]: Good suggestion, thanks!

**Pull request merged**